AMD SoundWire SOF generic driver changes for ACP 6.3 platform

vijendarmukunda opened the pull request:
This patch series is revised version of https://github.com/thesofproject/linux/pull/4817.

Compared to previous patch series, Below are the major changes.
 - Define common Soundwire machine driver soc ops
 - Move common Soundwire helper functions
 - Move Soundwire codec helper functions 

Not able to move all codec helper functions(like jack) as they are dependent on machine driver private data structures which will be different for Intel & AMD platforms.
If partly codec helper functions made common, i think it's not good a choice to make few of codec helper functions as generic and few codec helper functions as platform specific.

In next series, will work on DAI parsing logic.

I do have still doubts on copyrights usage as I have moved codec helper files from intel/boards -> sdw_utils folder.
Please help to provide suggestion on correct copyright usage.

vijendarmukunda committed:
ASoC: amd: acp: add soundwire machines for acp6.3 based platform

Add Soundwire machines for acp6.3 based platform.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: SOF: amd: add alternate machines for acp6.3 based platform

Add SoundWire machines as alternate machines for acp6.3 based platform.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw-utils: move soundwire machine driver soc ops

Move Intel SoundWire generic machine driver soc ops to common place
so that it can be used by other platform machine driver.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: move soundwire machine driver common structures

Move intel generic SoundWire machine driver common structures to SoundWire
utils header file. These structures will be used in other platform
SoundWire machine driver code.
Rename macro 'SOF_SDW_MAX_DAI_NUM' as 'SOC_SDW_MAX_DAI_NUM'.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda commented:
@charleskeepax @sunilkumardommati @bhiregou please help to review

plbossart reviewed (commented):
@vijendarmukunda I started an Intel review of the copyright changes. We'll look at the code after getting guidance on the copyright.

plbossart commented:
@vijendarmukunda the preference from Intel would be to add a comment 

````
/* This file incorporates work covered by the following copyright notice: Copyright (c) 2020 Intel Corporation */
````  
below the new AMD copyright notice. (the year may vary depending on files)

charleskeepax commented on sound/soc/sdw_utils/soc_sdw_utils.h:
```diff
@@ -12,6 +12,7 @@
 #include <sound/soc-acpi.h>
 
 #define SOF_SDW_MAX_DAI_NUM		8
+#define RT1316_MAX_BQ_REG		580
```
This is pretty RT specific feels like it should be in some RT header rather than the common util header.

vijendarmukunda replied:
will move it.

charleskeepax commented on sound/soc/sdw_utils/sof_sdw_amp_coeff_tables.h:
```diff
@@ -0,0 +1,300 @@
+ */
+
+/*
+ *  sof_sdw_amp_coeff_tables.h - related coefficients for amplifier parameters
```
This file name should indicate it is RT specific as well.

vijendarmukunda replied:
Agreed. Will rename the file name.

charleskeepax commented on sound/soc/sdw_utils/soc_sdw_dmic.c:
```diff
@@ -1,14 +1,14 @@
 // SPDX-License-Identifier: GPL-2.0-only
-// Copyright (c) 2020 Intel Corporation
+// Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved.
```
Pretty sure one should be keeping both copyrights here, rather than replacing the existing one, but hopefully someone with more legal knowledge can comment.

vijendarmukunda commented:
> @vijendarmukunda the preference from Intel would be to add a comment
> 
> ```
> /* This file incorporates work covered by the following copyright notice: Copyright (c) 2020 Intel Corporation */
> ```
> 
> below the new AMD copyright notice. (the year may vary depending on files)

will update the copyright

plbossart commented on sound/soc/sdw_utils/Kconfig:
```diff
@@ -0,0 +1,4 @@
+# SPDX-License-Identifier: GPL-2.0-only
+config SND_SOC_SDW_UTILS
+	tristate "SoundWire ASoC common helper functions"
```
It's my understanding that when we add a "string" then the option becomes visible at the make menuconfig level. That option should really be selected only, the user or distro shouldn't try to enable it.

vijendarmukunda replied:
will change it. Is it okay to add default value as "SOUNDWIRE" for this Kconfig option?

plbossart commented on sound/soc/sdw_utils/soc_sdw_utils.c:
```diff
@@ -0,0 +1,149 @@
+// Copyright(c) 2024 Advanced Micro Devices, Inc.
+/*
+ *  soc-sdw-utils.c - common SoundWire helper functions
+ *  Author: Vijendar Mukunda <Vijendar.Mukunda@amd.com>
```
I personally no longer add an "Author". The commit logs are more than enough to give you credit for your work.

vijendarmukunda replied:
Will drop it.

plbossart commented on sound/soc/sdw_utils/soc_sdw_utils.h:
```diff
@@ -9,6 +9,40 @@
+	const int part_id;
+	const int version_id;
+	const char *codec_name;
+	int amp_num;
```
@bardliao can you remind me why this field is required?

plbossart commented on sound/soc/sdw_utils/soc_sdw_utils.h:
```diff
@@ -9,6 +9,40 @@
+	const u8 acpi_id[ACPI_ID_LEN];
+	const bool ignore_pch_dmic;
+	const struct snd_soc_ops *ops;
+	struct sof_sdw_dai_info dais[SOF_SDW_MAX_DAI_NUM];
```
should this be const?

vijendarmukunda replied:
No. Because in dai info structure , we have field assignment (rtd_init_done).

plbossart commented on sound/soc/sdw_utils/soc_sdw_utils.h:
```diff
@@ -9,6 +9,40 @@
+	const char *codec_name;
+	int amp_num;
+	const u8 acpi_id[ACPI_ID_LEN];
+	const bool ignore_pch_dmic;
```
@bardliao we probably need to clean this up, e.g. RT722, RT1712 and RT1713 do not that this boolean set, and it's redundant with the dai_type SOF_SDW_DAI_TYPE_MIC.

vijendarmukunda replied:
As per our understanding, In below check, dmic_num > 0 is enough to create DMIC dai link for PCH DMIC.

   /* dmic */
        if (dmic_num > 0) {
                if (ctx->ignore_pch_dmic) {
                        dev_warn(dev, "Ignoring PCH DMIC\n");
                } else {
                        ret = create_dmic_dailinks(card, &dai_links, &be_id);
                        if (ret)
                                goto err_end;
                }
        }
I think we can drop ignore_pch_dmic references.  Please confirm, I will make the changes as part of this patch series.

plbossart replied:
@bardliao do you remember why we needed this ignore_pch_dmic?

plbossart commented on sound/soc/intel/boards/sof_sdw_common.h:
```diff
@@ -75,7 +75,7 @@ struct mc_private {
 	unsigned int sdw_pin_index[SDW_MAX_LINKS];
 	bool append_dai_type;
-	bool ignore_pch_dmic;
+	bool ignore_int_dmic;
```
int as 'intel' or 'internal'?

I am not sure if this is needed in the first place, we have the DMIC dai type that should be used instead.

vijendarmukunda replied:
its 'internal'. will check

plbossart commented on sound/soc/sdw_utils/soc_sdw_utils.c:
```diff
@@ -145,5 +145,86 @@ void asoc_sdw_shutdown(struct snd_pcm_substream *substream)
+
+	return true;
+}
+EXPORT_SYMBOL_NS(is_unique_device, SND_SOC_SDW_UTILS);
```
even with a name space, it's probably best to have a prefix.

vijendarmukunda replied:
will add prefix to function name

plbossart commented on sound/soc/intel/boards/sof_sdw.c:
```diff
@@ -1462,7 +1462,7 @@ static int create_dmic_dailinks(struct snd_soc_card *card,
 				   0, 1, // DMIC only supports capture
 				   "DMIC01 Pin", "dmic-codec", "dmic-hifi",
-				   sof_sdw_dmic_init, NULL);
+				   asoc_sdw_dmic_init, NULL);
```
Erm, not sure about this one. The Intel DMIC is really really specific, and we can have two DAIs for the same interface. Probably best to leave in the Intel parts.

vijendarmukunda replied:
This seems to be generic. Even though you have two DAI's . you are calling this function only once. Please correct me, if my understanding is wrong.

plbossart commented on sound/soc/sdw_utils/soc_sdw_rt_dmic.c:
```diff
@@ -51,4 +50,4 @@ int rt_dmic_rtd_init(struct snd_soc_pcm_runtime *rtd)
 	return 0;
 }
-MODULE_IMPORT_NS(SND_SOC_INTEL_SOF_BOARD_HELPERS);
+EXPORT_SYMBOL_NS(rt_dmic_rtd_init, SND_SOC_SDW_UTILS);
```
best to have a sdw_ prefix or something that clearly identifies where the helper came from.

vijendarmukunda replied:
will add sdw_ prefix.

plbossart commented on sound/soc/intel/boards/sof_sdw_rt_amp.c:
```diff
@@ -321,3 +115,4 @@ int sof_sdw_rt_amp_init(struct snd_soc_card *card,
 
 	return 0;
 }
+MODULE_IMPORT_NS(SND_SOC_SDW_UTILS);
```
@bardliao do you remember if we have any SoundWire platforms where the rt1308 is used in I2S mode?

This was an early TGL RVP IIRC, but I think we stopped supporting this, didn't we?

plbossart commented on sound/soc/amd/acp/amd-acp63-acpi-match.c:
```diff
@@ -0,0 +1,89 @@
+		.link_mask = BIT(0) | BIT(1),
+		.links = acp63_4_in_1_sdca,
+		.drv_name = "amd_sof_sdw",
+		.sof_tplg_filename = "sof-acp_6_3-rt711-l0-rt1316-l0-rt714-l1.tplg",
```
the naming conventions prevent you from showing that you have two rt1316 on link0.

It's not really a problem for playback, but if you want to get any sort of feedback then you need to know if the topology has 1 or 2 feedback links.

Note that Intel would have the same problem but currently we typically have two amps on two links, so we use e.g. rt1316-l12- to show the difference with rt1316-l1 if there's a single amp.

vijendarmukunda replied:
Currently we don't have any OEM platform enabled with Soundwire solution. we are evaluating using codec eval boards. Currently AMD platforms doesn't support multi-link aggregation like Intel. This is kind of initial drop which enables basic SoundWire solution on AMD platforms. we will revisit the naming convention later.

plbossart commented on sound/soc/amd/acp/acp-sdw-sof-mach.c:
```diff
@@ -0,0 +1,1084 @@
+	if (SOF_JACK_JDSRC(sof_sdw_quirk))
+		dev_dbg(dev, "quirk realtek,jack-detect-source %ld\n",
+			SOF_JACK_JDSRC(sof_sdw_quirk));
+	if (sof_sdw_quirk & SOF_SDW_FOUR_SPK)
```
remove

vijendarmukunda replied:
We want to keep it. We have already validated TI codecs with 4 mono speakers. So Four speaker aggregation is valid in this case.

plbossart commented on sound/soc/amd/acp/acp-sdw-sof-mach.c:
```diff
@@ -0,0 +1,1084 @@
+		dev_dbg(dev, "quirk SOF_SDW_FOUR_SPK enabled\n");
+	if (sof_sdw_quirk & SOF_SDW_ACP_DMIC)
+		dev_dbg(dev, "quirk SOF_SDW_ACP_DMIC enabled\n");
+	if (sof_sdw_quirk & SOF_SDW_NO_AGGREGATION)
```
also remove

vijendarmukunda replied:
we are using SOF_SDW_NO_AGGREGATION flag in multiple places. We are going to change DAI parsing logic . During that clean up we will change the code.

plbossart commented on sound/soc/amd/acp/acp-sdw-sof-mach.c:
```diff
@@ -0,0 +1,1084 @@
+		},
+		.dai_num = 1,
+	},
+};
```
Oh very interesting. 

Since there's nothing Intel or AMD-specific here, should those tables be in the helpers as well?

This is literally a copy-paste of the same codec-related information.

vijendarmukunda replied:
We have multiple reasons to keep separate tables in AMD machine driver.
1)we want to add support for codecs which we have validated so far on our platforms. If some one by looking at code get to know what are the SoundWire Codecs supported/ functional validation done on AMD platforms.
2) we haven't exported all codec helper functions(For example Jack related which dependency on machine driver private data structure) which is platform specific these tables can't be made generic.
3) Apart from above limitation, As we haven't validated Cirrus and Maxim codecs in Linux environment, we haven't moved codec helper functions related to Cirrus and Maxim codecs

plbossart replied:
you could add another filter in the AMD parts that ignores the parts you haven't validated. That would be a much better indicator of what works and what doesn't. Duplicating the baseline table is not really the best way to progress. There will be other codec vendors and other parts showing up in 2025, we need to reduce the amount of duplication otherwise it'll become unmanageable.

vijendarmukunda replied:
I do agree. Currently we have evaluated most of RTK codecs(RT711, RT712, RT713, RT1316, RT722) and TI Soundwire codecs.  We are yet to validate  Cirrus Codecs.  If we keep common codec tables, then all codec helper functions should be made generic.
The problem here is mc_private structure is not common for Intel and AMD platforms.
Currently jack related helper functions are not made generic.eg. sof_sdw_rt_sdca_jack_init().
Please suggest how can we address this issue, if we want to make codec tables common?

plbossart replied:
The existing private structure isn't really Intel-specific

````
struct mc_private {
	struct snd_soc_jack sdw_headset;
	struct sof_hdmi_private hdmi;
	struct device *headset_codec_dev; /* only one headset per card */
	struct device *amp_dev1, *amp_dev2;
	/* To store SDW Pin index for each SoundWire link */
	unsigned int sdw_pin_index[SDW_MAX_LINKS];
	bool append_dai_type;
	bool ignore_pch_dmic;
};
````

the ignore_pch_dmic could be removed, and then it's fairly straightforward. We could move some part to a truly private mv_vendor_specific structure if needed, and treat it as a (void *).

vijendarmukunda replied:
yes, You are correct. If we move some part to truly mv_vendor_structure, it may solve the issue where we can keep common mc_private structure in this case.

plbossart commented on sound/soc/amd/acp/acp-sdw-sof-mach.c:
```diff
@@ -0,0 +1,1084 @@
+	}
+
+	return 0;
+}
```
that looks like copy-paste as well.

vijendarmukunda replied:
The reason for not able to export this function is codec info table is local to this file.

plbossart commented on sound/soc/amd/acp/acp-sdw-sof-mach.c:
```diff
@@ -0,0 +1,1084 @@
+			"SDW%d-PIN%d-PLAYBACK",
+			"SDW%d-PIN%d-CAPTURE",
+			"SDW%d-PIN%d-PLAYBACK-%s",
+			"SDW%d-PIN%d-CAPTURE-%s",
```
that looks different

vijendarmukunda replied:
yes

plbossart commented on sound/soc/amd/acp/acp-sdw-sof-mach.c:
```diff
@@ -0,0 +1,1084 @@
+				break;
+			case SDW_DMIC_DAI_ID:
+				cpu_pin_id = SW0_AUDIO2_RX;
+				break;
```
and that as well, so that clearly belongs in the AMD parts.

vijendarmukunda replied:
Correct. SDW DAI link creation logic is different from Intel.

plbossart commented on sound/soc/amd/acp/acp-sdw-sof-mach.c:
```diff
@@ -0,0 +1,1084 @@
+
+	WARN_ON(dai_links != card->dai_link + card->num_links);
+	return 0;
+}
```
that looks fairly generic again, so wondering if we can reuse with some sort of mask that would tell which interfaces to create a link for.

vijendarmukunda replied:
We are going to change the dai creation logic and will push it as a separate PR. This PR is mainly targeted to pull out generic functions and define common codec helper functions.

plbossart commented on sound/soc/amd/acp/acp-sdw-sof-mach.c:
```diff
@@ -0,0 +1,1084 @@
+		}
+	}
+	return ret;
+}
```
clearly generic?

vijendarmukunda replied:
As codec_info_list structure is not exported that's why we left this function as static.

plbossart reviewed (commented):
I think it's a good start @vijendarmukunda 

Reviewing all this code helps identify all stuff that could be simplified, e.g. the ignore_pch_dmic.

moving the codec stuff is the right thing to so, but see below the list of codec_info could be moved to generic parts, there's really nothing AMD or Intel specific about a Cirrus or Realtek or TI part.

There are also parts that could be moved to the more generic helpers, there's too many remaining copy-pasted files.

but overall good direction, it'll make our esteemed ALSA maintainers happy.

vijendarmukunda commented:
> @vijendarmukunda the preference from Intel would be to add a comment
> 
> ```
> /* This file incorporates work covered by the following copyright notice: Copyright (c) 2020 Intel Corporation */
> ```
> 
> below the new AMD copyright notice. (the year may vary depending on files)

will update the copyright.

vijendarmukunda committed:
ASoC: intel/sdw_utils: rename ignore_pch_dmic variable name

Rename 'ignore_pch_dmic' variable name as 'ignore_internal_dmic' to make
it more generic across different platforms to represent internal dmic.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: move dmic codec helper functions

Move generic dmic codec helper functions implementation to
sdw_utils folder sothat these functions can be used by other platform
machine drivers.
Rename the helper function 'sof_sdw_dmic_init' as 'asoc_sdw_dmic_init'
as this helper function will be used by SOF and legacy(No DSP) stacks.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: move rt712 sdca helper functions

Move RT712 SDCA codec helper file to sdw_utils folder.
Rename 'rt712_spk_rtd_init' function as 'rt712_sdw_spk_rtd_init'.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: move rt722 sdca helper functions

Move RT722 SDCA codec helper file to sdw_utils folder to make it generic.
Rename helper function with _sdw tag.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: move soundwire machine driver private data structure

Move Soundwire generic machine driver private data structure to
soc_sdw_utils.h file so that it can be used by other platform
machine driver.

Separate vendor specific machine driver data from common machine driver
private data structure as a void pointer and assign the vendor specific
structure in mc_probe() call.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: move rt5682 codec helper functions

Move rt5682 sdw codec helper functions to common place holder to make it
generic. Rename 'rt5682_rtd_init' function as 'rt5682_sdw_rtd_init'.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: move rt700, rt711 codec helper functions

Move RT700 and RT711 Soundwire Codec helper functions to common
place holder so that it can be used by other platform machine driver.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: move rtk dmic helper functions

Move rtk SoundWire dmic helper functions implementation to sdw_utils
folder to make it generic. Rename dmic helper functions by adding _sdw
tag.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: move soundwire dai type macros

Move SoundWire dai type macros to common header file(soc_sdw_util.h) and
rename the macros to make it generic.It will be used by other platform
machine driver.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: move machine driver dai link helper functions

Move machine driver dai link helper functions to common place holder,
so that it can be used by other platform machine driver.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: soc-acpi: add pci revision id field in mach params structure

Few IP's may have same PCI vendor id and device id and
different revision id. Add revision id field to the
'snd_soc_acpi_mach_params' so that using this field platform
specific implementation can be added in machine driver.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: SOF: amd: update mach params subsystem_rev variable

Add pci_rev variable in acp sof driver private data structure and assign
this value to mach_params structure subsystem_rev variable.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: amd: acp: add sof based soundwire generic machine driver

Add sof based Soundwire generic driver for amd platforms.
Currently support added for ACP 6.3 based platforms.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: move soundwire machine driver helper functions

Move below Intel SoundWire machine driver helper functions to
soc_sdw_utils.c file so that it can be used by other platform machine
driver.
- is_unique_device()
- get_codec_name()
- get_codec_dai_by_name()
Rename the functions by adding _sdw tag.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: move soundwire codec_info_list structure

SoundWire 'codec_info_list' structure is not a platform specific one.
Move codec_info_list structure to common file soc_sdw_utils.c.
Move codec helper functions which uses codec_info_list structure to common
place holder and rename the function by adding _sdw tag. This will allow
to use 'codec_info_list' structure and it's helper functions in other
platform machine driver.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: sdw_utils: refactor sof_sdw_card_late_probe function

Refactor sof_sdw_card_late_probe() function and derive a generic
function soc_sdw_card_late_probe() function which can be used by
SoundWire generic machine driver.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: move rtk jack common helper functions

Move RTK codec jack common helper functions to common place holder
(sdw_utils folder) to make it generic so that it will be used by
other platform machine driver code.
Rename codec helper functions by adding _sdw tag.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: move rtk amp codec helper functions

Move RTK amp codec helper functions related implementation to common
place holder to make it generic so that these helper functions will be
used by other platform machine driver modules.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: move maxim codec helper functions

Move maxim codec helper functions to common place holder so that
it can be used by other platform machine driver.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: rename codec_info and dai_info structures names

To make it generic, rename structure 'sof_sdw_codec_info' as
'soc_sdw_codec_info' and 'sof_sdw_dai_info' as 'soc_sdw_dai_info'.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: move dai id common macros

Move dai id common macros from intel SoundWire generic driver to
soc_sdw_utils.h file so that it can be used by other platform machine
driver.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda committed:
ASoC: intel/sdw_utils: move cirrus soundwire codec helper functions

To make it generic, move Cirrus Soundwire codec helper functions to
common place holder so that it can be used by other platform machine
driver. Rename the codec helper functions.

Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>

vijendarmukunda commented:
Changes Since V1:

1) Moved all SoundWire codec helper functions to sdw_utils folder.
2) Updated copyright statement
3) Moved codec_info_list structure and its corresponding helper functions.
4) Moved common Soundwire codec helper functions.
5) Moved Machine driver dai link helper functions.
6) Refactored late probe() function
7) Add pci revision id field in mach_params structure and used it in amd sof driver stack.
8) Renamed all common functions with _sdw tag.
9) Rename structure names and macros to make it generic.

vijendarmukunda commented:
observing merge conflicts will rebase and log the changes in new pr.
